doc/adr/0007-marconi-indexer.rst

```diff
@@ -0,0 +1,118 @@
+.. _marconi_indexer:
+
+ADR X: Marconi indexers
+=======================
+
+Date: 2022-08-05
+
+Authors
+-------
+
+Ometita Radu <radu.ometita@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Off-chain code needs access to indexed portions of the blockchain. Currently, we have a working solution in the form of the chain-index and PAB (which both index information). The big problem with the current solution is its lack of reuse (or modularity) capability.
+
+We attempt to fix that problem with Marconi where we currently have a generic indexer that stores volatile information in memory and blocks that have been fully committed (are old enough to guarantee that they will not be rolled back). The K blockchain parameter represents the number of slots after which a block can no longer be rolled back.
```

Comment:
blocks that have been fully committed -> blocks that have been fully committed on disk